libraries/common/stream.effekt

```diff
@@ -0,0 +1,417 @@
+module stream
+
+import array
+import bytearray
+import io/filesystem
+import io/error
+
+
+effect emit[A](value: A): Unit
+
+def for[A, R] { stream: () => R / emit[A] } { action: A => Unit }: R =
+  try {
+    stream()
+  } with emit[A] { value =>
+    resume(action(value))
+  }
+
+def for[A] { stream: () => Unit / emit[A] } { action: A => Unit }: Unit =
+  for[A, Unit]{stream}{action}
+
+def continually[A] { action: () => A }: Nothing / emit[A] = {
+  def go(): Nothing = {
+    do emit(action())
+    go()
+  }
+  go()
+}
+
+
+def each[A](list: List[A]): Unit / emit[A] =
+  list match {
+    case Nil() => ()
+    case Cons(head, tail) =>
+      do emit(head)
+      each(tail)
+  }
+
+def each[T](array: Array[T]): Unit / emit[T] = {
+  val n = array.size
+  def go(i: Int): Unit = {
+    if (i < n) {
+      do emit(array.unsafeGet(i))
+      go(i + 1)
+    }
+  }
+  go(0)
+}
+
+def each(bytes: ByteArray): Unit / emit[Byte] = {
+  val n = bytes.size
+  def go(i: Int): Unit = {
+    if (i < n) {
+      do emit(bytes.unsafeGet(i))
+      go(i + 1)
+    }
+  }
+  go(0)
+}
+
+// not option
+// not dequeue
+// not queue
+// not result
+// not seq
+
+
+effect stop(): Nothing
```

Comment:
We should have some common best practice on where to define effects. In the middle of files, all at the top?